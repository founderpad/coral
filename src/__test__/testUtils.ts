// import { ThemeProvider } from "my-ui-lib"
// import { TranslationProvider } from "my-i18n-lib"
// import defaultStrings from "i18n/en-x-default"
import { render } from '@testing-library/react';
import { JSXElementConstructor, ReactElement } from 'react';

const Providers = ({ children }: { children: any }) => {
	return children;
	// return (
	//   <ThemeProvider theme="light">
	//     <TranslationProvider messages={defaultStrings}>
	//       {children}
	//     </TranslationProvider>
	//   </ThemeProvider>
	// )
};

const customRender = (
	ui: ReactElement<any, string | JSXElementConstructor<any>>,
	options = {}
): any => render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
