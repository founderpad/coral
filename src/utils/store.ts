import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { persistReducer } from 'redux-persist';
import { PersistConfig } from 'redux-persist/es/types';
import storage from 'redux-persist/lib/storage';
import rootReducer from './reducer';

/**
 * Create the Redux store
 *
 * @author jlee
 */

// const isProd = process.env.NODE_ENV !== 'development';

/**
 * Persist storage configuration set up
 * @key[root]
 * @storage[storage]
 */
const persistConfig: PersistConfig<any> = {
	key: 'root',
	storage
};

const middleware = [...getDefaultMiddleware({ serializableCheck: false })];

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
	reducer: persistedReducer,
	middleware
	// devTools: isProd ?? false
});

// export { StoreProvider }

export default store;
