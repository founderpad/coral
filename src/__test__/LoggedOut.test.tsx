import LoggedOut from 'pages/app/loggedout';
import { Provider } from 'react-redux';
import store from 'utils/store';
import { render } from '__test__/testUtils';

const setup = () =>
	render(
		<Provider store={store}>
			<LoggedOut />
		</Provider>
	);

describe('Logged out page', () => {
	it('matches snapshot', () => {
		const { asFragment } = setup();
		expect(asFragment()).toMatchSnapshot();
	});

	it('should render LoggedOut page', () => {
		const { getByRole } = setup();
		expect(getByRole('link')).toBeInTheDocument();
	});
});
