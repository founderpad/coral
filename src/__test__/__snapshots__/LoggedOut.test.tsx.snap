// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logged out page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-wyca3c"
  >
    <div
      class="css-1n3ziv8"
    >
      <div
        class="chakra-stack css-1hgee5k"
      >
        <img
          alt="logo"
          class="chakra-image css-1b7p44u"
          src="/logo.svg"
        />
        <h2
          class="chakra-heading css-1zp44w"
        >
          You've logged out successfully
        </h2>
        <p
          class="chakra-text css-1mjwwnw"
        />
        <a
          class="chakra-link css-1d7bnbh"
          href="/login"
          title="Log in"
        >
          Click here to log in
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
